azure/activity_logs_monitoring/index.js

```diff
@@ -102,6 +102,10 @@ function handleJSONArrayLogs(sender, context, logs, logsType) {
                 return;
             }
         }
+        // If the message is a buffer object, the data type has been set to binary.
+        if (Buffer.isBuffer(message)) {
+            message = JSON.parse(message.toString());
+        }
```

Comment:
I think it could be confusing that this is considered json-array when it's not actually a json array. What do you think about some kind of check in this conditional below

```
    if (typeof logs[0] === 'object') {
       return JSON_ARRAY;
    }	
```
checking if if it is a buffer object, and then returning a logType of BUFFER_ARRAY? then we also don't need to check if it is a buffer for every message, just that it is only for the first

Reply:
Good idea. I've made this change in the latest refactor. Lemme know what you think!